src/librustc/traits/error_reporting.rs

```diff
@@ -1086,6 +1087,11 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
 
             // already reported in the query
             ConstEvalFailure(err) => {
+                if let ErrorHandled::TooGeneric = err {
+                    // Silence this error, as it can be produced during intermediate steps
+                    // when a constant is not yet able to be evaluated (but will be later).
```

Comment:
So this is a bit like `assert_reported`, except `delay_span_bug`, and it's missing the exhaustive pattern-matching:
https://github.com/rust-lang/rust/blob/117ceeba400e69aa7b8ba5155acd1c8c0cdf7f7f/src/librustc/mir/interpret/error.rs#L30-L40

But also, I'm not sure `ConstEvalFailure` should exist anymore. Or maybe `ConstEvalFailure(ErrorHandled::TooGeneric)` should never be created (since it's not really an error).

In any case, nowadays the WF for constants is just a side-effect, as opposed to relying on error propagation, so we can probably get rid of `ConstEvalFailure`.

Reply:
Disregard previous comment, I'm very wrong.